### PR TITLE
feat(rds): database instance - auto scale allocated storage

### LINKED ISSUE #6666: [RDS] Support Storage Autoscaling option on Database Instance
Add support for database storage autoscaling option for RDS Database Instance in AWS CDK.

### Use Case

I would like to leverage storage autoscaling for my CDK deployed RDS database: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PIOPS.StorageTypes.html#USER_PIOPS.Autoscaling

### Proposed Solution

Add a `maxAllocatedStorage` option for `@aws-cdk/aws-rds.DatabaseInstance` now that https://github.com/aws-cloudformation/aws-cloudformation-coverage-roadmap/issues/72 is shipped.

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by nirvana124**
## Commit Message
feat(rds): database instance - auto scale allocated storage (#6788)

fixes #6666 
## End Commit Message

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-rds/README.md [hunk 1]
```diff
@@ -76,6 +76,21 @@ new DatabaseInstanceReadReplica(stack, 'ReadReplica', {
...
+```ts
+const instance = new DatabaseInstance(stack, 'Instance', {
+    engine: rds.DatabaseInstanceEngine.ORACLE_SE1,
+    instanceClass: ec2.InstanceType.of(ec2.InstanceClass.BURSTABLE2, ec2.InstanceSize.SMALL),
+    masterUsername: 'syscdk',
+    vpc,
+    maxAllocatedStorage: 200
+});
+```
+
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/README.md]: Can we move this to line 60 instead?

File: packages/@aws-cdk/aws-rds/README.md [hunk 2]
```diff
@@ -76,6 +76,21 @@ new DatabaseInstanceReadReplica(stack, 'ReadReplica', {
     vpc
 });
 ```
+
+To use the storage auto scaling option of RDS you can specify the maximum allocated storage.
+This is the upper limit to which RDS can automatically scale the storage. More info can be found
+[here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-maxallocatedstorage)
+Example for max storage configuration:
+```ts
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/README.md]: Normally, we leave a new line before and after a code block.

```suggestion

```ts
```

File: packages/@aws-cdk/aws-rds/README.md [hunk 3]
```diff
@@ -76,6 +76,21 @@ new DatabaseInstanceReadReplica(stack, 'ReadReplica', {
     vpc
 });
 ```
+
+To use the storage auto scaling option of RDS you can specify the maximum allocated storage.
+This is the upper limit to which RDS can automatically scale the storage. More info can be found
+[here](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-maxallocatedstorage)
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/README.md]:
```suggestion
[here](https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PIOPS.StorageTypes.html#USER_PIOPS.Autoscaling)
```

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 4]
```diff
@@ -478,6 +478,12 @@ export interface DatabaseInstanceNewProps {
    * @default RemovalPolicy.Retain
    */
   readonly removalPolicy?: RemovalPolicy
+
+  /**
+   * Upper limit to which RDS can scale the storage
+   * @default - No default value
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/instance.ts]: Explain what this default actually means. Does it mean that the RDS instance does not scale, or scale as much as the underlying hardware allows it?

File: packages/@aws-cdk/aws-rds/lib/instance.ts [hunk 5]
```diff
@@ -478,6 +478,12 @@ export interface DatabaseInstanceNewProps {
    * @default RemovalPolicy.Retain
    */
   readonly removalPolicy?: RemovalPolicy
+
+  /**
+   * Upper limit to which RDS can scale the storage
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/instance.ts]:
```suggestion
   * Upper limit to which RDS can scale the storage
   * @see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/USER_PIOPS.StorageTypes.html#USER_PIOPS.Autoscaling
```

Comment (nija-at) [inline packages/@aws-cdk/aws-rds/lib/instance.ts]: units?

File: packages/@aws-cdk/aws-rds/test/test.instance.ts [hunk 6]
```diff
@@ -716,6 +716,52 @@ export = {
...
+
+    expect(stack).to(haveResource('AWS::RDS::DBInstance', {
+      DBSnapshotIdentifier: 'my-snapshot',
+      MaxAllocatedStorage: 200
+    }));
+
+    test.done();
+  },
+
+  'create a DB instance with maxiumum allocated storage'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/test/test.instance.ts]:
```suggestion
  'create a DB instance with maximum allocated storage'(test: Test) {
```

File: packages/@aws-cdk/aws-rds/test/test.instance.ts [hunk 7]
```diff
@@ -716,6 +716,52 @@ export = {
       }), /timezone property can be configured only for Microsoft SQL Server/);
     });
 
+    test.done();
+  },
+
+  'create an instance from snapshot with maxiumum allocated storage'(test: Test) {
```
Comment (nija-at) [inline packages/@aws-cdk/aws-rds/test/test.instance.ts]:
```suggestion
  'create an instance from snapshot with maximum allocated storage'(test: Test) {
```

File: packages/@aws-cdk/aws-rds/README.md (hunk 1 above)
Comment (nirvana124) [reply packages/@aws-cdk/aws-rds/README.md]: Moved it to 61.

**Pull request merged**